src/server/room-controller.spec.js

```diff
@@ -649,6 +650,67 @@ describe('room-controller', () => {
     });
   });
 
+  describe('handleDeleteRooms', () => {
+    const userJacky = [{ _id: uniqueId.create(), email: 'jacky@test.com' }];
+    const userClare = [{ _id: uniqueId.create(), email: 'clare@test.com' }];
+    const userDrake = [{ _id: uniqueId.create(), email: 'drake@test.com' }];
+    const roomA = { _id: uniqueId.create(), name: 'Room A', access: ROOM_ACCESS_LEVEL.private, members: [{ userId: userJacky._id }, { userId: userClare._id }] };
+    const roomB = { _id: uniqueId.create(), name: 'Room B', access: ROOM_ACCESS_LEVEL.private, members: [{ userId: userClare._id }, { userId: userDrake._id }] };
+    const roomC = { _id: uniqueId.create(), name: 'Room C', access: ROOM_ACCESS_LEVEL.public, members: [{ userId: userJacky._id }, { userId: userDrake._id }] };
+
+    beforeEach(done => {
+      req = httpMocks.createRequest({
+        protocol: 'https',
+        headers: { host: 'educandu.dev' },
+        query: { ownerId: user._id, access: ROOM_ACCESS_LEVEL.private }
+      });
+      req.user = user;
+
+      res = httpMocks.createResponse({ eventEmitter: EventEmitter });
+      res.on('end', done);
+
+      roomService.getRoomsOwnedByUser.resolves([roomA, roomB]);
```

Comment:
It's a little confusing that here this resolves with the users of `userJacky` but the `req.query.onwerId` is not set up to the same user.
I think it would be good to include `withArgs` here as well, to resolve per user. That would basically point to the correction needed in the `req` setup.

Reply:
I added `withArgs`, and also detected that it should resolve to all three rooms (I added `roomC` later), as they are all owned by `dagobert-the-third`

Reply:
This guy, he owns too much 🤔